extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala

```diff
@@ -98,14 +109,14 @@ private[authz] object AuthZUtils {
     plan.nodeName == "DataSourceV2Relation" && plan.resolved
   }
 
-  def getDatasourceV2Identifier(plan: LogicalPlan): Option[TableIdentifier] = {
-    // avoid importing DataSourceV2Relation for Spark version compatibility
-    val identifier = getFieldVal[Option[AnyRef]](plan, "identifier")
-    identifier.map { id =>
-      val namespaces = invoke(id, "namespace").asInstanceOf[Array[String]]
-      val table = invoke(id, "name").asInstanceOf[String]
-      TableIdentifier(table, Some(quote(namespaces)))
-    }
+  def getDatasourceV2Identifier(plan: LogicalPlan): Option[Identifier] = {
+    getFieldVal[Option[Identifier]](plan, "identifier")
+  }
+
+  def getTableIdentifierFromIdentifier(id: Identifier): TableIdentifier = {
+    val namespaces = invoke(id, "namespace").asInstanceOf[Array[String]]
```

Comment:
what not call id.namespace() directly?

Reply:
ok, it's a legacy usage from 1.6.0. let me fix it.